x-pack/test/functional/page_objects/endpoint_page.ts

```diff
@@ -10,6 +10,24 @@ export function EndpointPageProvider({ getService }: FtrProviderContext) {
   const testSubjects = getService('testSubjects');
 
   return {
+    /**
+     * Finds the Data Grid with the given `selector` (test subject) and returns
+     * back an array containing the grid's header column text
+     *
+     * @param selector
+     * @returns Promise<string[]>
+     */
+    async dataGridHeaderVisibleText(selector: string) {
+      const $ = await (await testSubjects.find(selector)).parseDomContent();
+      return $('.euiDataGridHeaderCell__content')
```

Comment:
Consider using data-subj. Coupling your test to an internal property of the implementation like this is more likely to be brittle.